eclair-core/src/main/scala/fr/acinq/eclair/channel/Commitments.scala

```diff
@@ -222,14 +222,14 @@ case class Commitments(channelId: ByteVector32,
   val localChannelReserve: Satoshi = if (channelFeatures.hasFeature(Features.DualFunding)) {
     (capacity / 100).max(remoteParams.dustLimit)
   } else {
-    remoteParams.requestedChannelReserve_opt.getOrElse(0 sat)
+    remoteParams.requestedChannelReserve_opt.get // this is guarded by a require() in Commitments
```

Comment:
This isn't correct, the `require` in Commitments only checks that when using dual funding, `requestedChannelReserve_opt` is `None`, not that when dual funding isn't used it is `Some`. And it feels to me that it actually shouldn't, when using zero-reserve it feels more natural to set `requestedChannelReserve_opt` to `None` than to `Some(0 sat)`, isn't it?

Reply:
Nice catch! Fixed in 6a4e972bd, and it looks better.